src/main/scala/is/hail/annotations/aggregators/RegionValueDownsampleAggregator.scala

```diff
@@ -0,0 +1,53 @@
+package is.hail.annotations.aggregators
+import is.hail.annotations.RegionValueBuilder
+import is.hail.annotations._
+import is.hail.expr.types.{TArray, TFloat64, TTuple, TString}
+import is.hail.stats.DownsampleCombiner
+
+object RegionValueDownsampleAggregator {
+  val typ = TArray(TTuple(Array(TFloat64(), TFloat64(), TArray(TString()))))
+}
+
+class RegionValueDownsampleAggregator(nDivisions: Int) extends RegionValueAggregator {
+  require(nDivisions > 0)
+
+  var combiner: DownsampleCombiner = new DownsampleCombiner(nDivisions)
+
+  def seqOp(region: Region, x: Double, xm: Boolean, y: Double, ym: Boolean, l: Long, lm: Boolean): Unit = {
+    if (!xm && !ym) {
+      val label = if (lm) null else SafeRow.read(TArray(TString()), region, l).asInstanceOf[IndexedSeq[String]]
+      combiner.merge(x, y, label)
+    }
+  }
+  
+  def combOp(agg2: RegionValueAggregator) = combiner.merge(agg2.asInstanceOf[RegionValueDownsampleAggregator].combiner)
+
+  def result(rvb: RegionValueBuilder) {
+    rvb.startArray(combiner.binToPoint.size)
+    combiner.binToPoint.foreach { case (_, (x, y, a)) =>
+      rvb.startTuple()
+      rvb.addDouble(x)
+      rvb.addDouble(y)
+      if (a != null) {
```

Comment:
If a is null, the array should be missing (not null).  You can do this by calling `rvb.addAnnotation(TArray(TString()), a)` and that will handle the case `a` is `null`.

Reply:
Looks like you didn't address thi sone.

Reply:
whoops, misread your earlier comment. Fixing now.